src/app/pages/directory-service/components/directory-services-config-form/active-directory-config/active-directory-config.component.ts

```diff
@@ -0,0 +1,365 @@
+        this.enableTrustedDomains.set(enableTrustedDomains ?? false);
+        if (!enableTrustedDomains) {
+          this.form.controls.ad_trusted_domains.setValue([]);
+        }
+        this.cdr.markForCheck();
+      });
+
+    // Watch for idmap backend changes
+    this.form.controls.idmap_backend.valueChanges
+      .pipe(untilDestroyed(this))
+      .subscribe((backend) => {
+        this.selectedIdmapBackend.set(backend as string | null);
+        this.updateIdmapBackendFields(backend as string | null);
+        this.cdr.markForCheck();
+      });
+
+    // Watch for domain changes to suggest Kerberos realm
+    this.form.controls.ad_domain.valueChanges
+      .pipe(untilDestroyed(this))
+      .subscribe((domain: string | null) => {
+        if (domain && typeof domain === 'string') {
+          const domainStr = domain.trim();
+          if (domainStr) {
+            // Suggest uppercase version of domain as Kerberos realm
+            const suggestedRealm = domainStr.toUpperCase();
+            this.kerberosRealmSuggested.emit(suggestedRealm);
+          } else {
+            this.kerberosRealmSuggested.emit(null);
+          }
+        } else {
+          this.kerberosRealmSuggested.emit(null);
+        }
+      });
+
+    // Watch for any form changes and emit refined configuration object
+    this.form.valueChanges
+      .pipe(untilDestroyed(this))
+      .subscribe(() => {
+        this.configurationDataChanged.emit(this.buildActiveDirectoryConfig());
+      });
+  }
+
+  onTrustedDomainsChanged(trustedDomains: unknown[]): void {
+    this.form.controls.ad_trusted_domains.setValue(trustedDomains, { emitEvent: false });
+    this.configurationDataChanged.emit(this.buildActiveDirectoryConfig());
+  }
+
+  private buildActiveDirectoryConfig(): ActiveDirectoryConfig | null {
+    const formValue = this.form.value;
+
+    if (!formValue.ad_hostname || !formValue.ad_domain) {
+      return null;
+    }
+
+    return {
+      hostname: formValue.ad_hostname as string,
+      domain: formValue.ad_domain as string,
+      use_default_domain: formValue.ad_use_default_domain ?? false,
+      enable_trusted_domains: formValue.ad_enable_trusted_domains ?? false,
+      idmap: this.buildActiveDirectoryIdmap(),
+      site: (formValue.ad_site as string) || null,
+      computer_account_ou: (formValue.ad_computer_account_ou as string) || null,
+      trusted_domains: this.buildTrustedDomains(formValue.ad_trusted_domains || []),
+    };
+  }
+
+  private buildActiveDirectoryIdmap(): PrimaryDomainIdmap | PrimaryDomainIdmapAutoRid {
+    const formValue = this.form.value;
+
+    if (formValue.ad_use_default_idmap || !formValue.idmap_backend) {
+      return {
+        builtin: {
+          name: null,
+          range_low: 90000001,
+          range_high: 100000000,
+        },
+        idmap_domain: {
+          name: null,
+          idmap_backend: IdmapBackend.Ad,
+          range_low: formValue.idmap_range_low || 10000,
```

Comment:
Do we need to have default values here? Aren't they coming from the child component?